app/pairNumberFromColor.js

```diff
@@ -0,0 +1,31 @@
+import { MajorColorNames, MinorColorNames } from "./colorInfo.js";
+
+export default function getPairNumberFromColor(pair)
+        {
+        let majorIndex = -1;
+        for (let i = 0; i < MajorColorNames.length; i++)
+            {
+                if (MajorColorNames[i] == pair.majorColor)
+                {
+                    majorIndex = i;
+                    break;
+                }
+            }
+
+        let minorIndex = -1;

```

Comment:
Notice the repeating pattern: initialize to -1, iterate and compare. Separate this into a function and call it in two places - to avoid the duplication.